SwiftUIBestPractices.md

```diff
@@ -0,0 +1,187 @@
+    @AppStorage(AppStorageKeys.keyTwo) private var propertyTwo = 0.75
+    @AppStorage(AppStorageKeys.keyThree) private var propeertyThree = 0.5
+   
+    @Binding private var sliderValue: Double
+    @Binding private var wasSessionPreviouslyRunning: Bool
+
+    @StateObject private var object = DeviceObserver()
+    
+    private let numberFormatter = NumberFormatter.fractionFormatter
+    private let wasSessionPreviouslyRunning: Bool
+		private let settingsViewModel: SettingsViewModel
+
+    // MARK: - SettingsView
+    
+    init(sliderValue: Binding<Double>, wasSessionPreviouslyRunning: Binding<Bool>, settingsViewModel: SettingsViewModel) {
+					self._sliderValue = sliderValue
+					self._wasSessionPreviouslyRunning = wasSessionPreviouslyRunning
+					self.settingsViewModel = settingsViewModel
+		}
+    
+    // MARK: - View
+
+    var body: some View {
+        List {...}
+        .listStyle(InsetGroupedListStyle())
+        .navigationTitle("Voice Activation")
+        .navigationBarTitleDisplayMode(.inline)
+        .onAppear {
+            startSessionIfNeeded()
+        }
+        .onDisappear {
+            stopSessionIfNeeded()
+        }
+        .toolbar {
+            ToolbarItem(placement: .primaryAction) {
+                PickerView()
+            }
+        }
+    }
+
+    // MARK: - SettingsView
+
+    private func startAudioSessionIfNeeded() {
+        guard !wasSessionPreviouslyRunning else {
+            return
+        }
+        
+        audioSession.start()
+    }
+    
+    private func stopAudioSessionIfNeeded() {
+        guard !wasSessionPreviouslyRunning else {
+            return
+        }
+        
+        audioSession.stop()
+    }
+}
+
+struct VoiceActivationSettingsView_Previews: PreviewProvider {
+    static var previews: some View {
+        NavigationView {
+            SettingsView(audioSession: AVAudioSession())
+        }
+    }
+}
+```
+
+# Property Wrappers grouped together
+
+`@StateObject` , `@State` and `@AppStorage` properties are marked as private, because they do not need to be mutated outside of their containing file. Property wrappers of the same type should be grouped together. 
+In the example below property wrappers are listed first followed by properties without property wrappers. 
+
+```swift 
+struct SettingsView: View {
+
+    @Binding private(set) var isDisplayingCertificateView: Bool
+    @ObservableObject private(set) var settingsViewModel: SettingViewModel
+    @AppStorage(AppStorageKeys.keyOne) private var propertyOne = 0.5
+    @AppStorage(AppStorageKeys.keyTwo) private var propertyTwo = 0.75
+    @AppStorage(AppStorageKeys.keyThree) private var propertyThree = 0.5
+    @StateObject private var object = DeviceObserver()
+    
+    private let numberFormatter = NumberFormatter.fractionFormatter
+    private let wasSessionPreviouslyRunning: Bool
+ }
+```
+
+# Initializing SwiftUI Views
+
+## Creating Initializers
+
+### A synthesized initializer
```

Comment:
It's unclear what these initializer sections are trying to convey with regard to guidelines. What guideline is being proposed for each of these, and why? In the example above, the introductory text is clear, that it's about ordering and what comes before the next thing, and the example helps to illustrate that. Similar text on each of these might help to clarify the points / guidelines.